hw02-arraylist/src/main/java/org/mlaptev/otus/MyArrayList.java

```diff
@@ -0,0 +1,449 @@
+package org.mlaptev.otus;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.List;
+import java.util.ListIterator;
+import java.util.NoSuchElementException;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class MyArrayList<T> implements List<T> {
+
+  private static final Logger logger = LogManager.getLogger(MyArrayList.class);
+
+  private int size;
+  private int capacity;
+  private T[] elements;
+
+  public MyArrayList(int capacity) {
+    this.size = 0;
```

Comment:
Absolutely! It is a very good point, I've added an additional check with throwing an exception in case when the capacity is less than zero. 
Please have a look!